File: endpoint-clob-websocket/template.js
const WebSocket = require("ws");
const fs = require("fs");

const tokenId = "90468297690119961729415596184451710609666058555177444095392763708049826586245";
const logFile = fs.createWriteStream("logs/log_orderbook.log", { flags: "a" });

const ws = new WebSocket("wss://ws-subscriptions-clob.polymarket.com/ws/market");

ws.onopen = () => {
  ws.send(JSON.stringify({
    type: "market",
    assets_ids: [tokenId]
  }));

//  setInterval(() => ws.send("PING"), 10000);

};

ws.onmessage = (event) => {
  const data = JSON.parse(event.data);
//  console.log(data);
  const timestamp = new Date().toISOString();
  const logEntry = `${timestamp}: ${JSON.stringify(data)}\n`;
  
  console.log(data);          
  logFile.write(logEntry);
};

File: endpoint-clob-websocket/template.py
import json
from datetime import datetime
from websocket import create_connection

#token_id = "90468297690119961729415596184451710609666058555177444095392763708049826586245"
token_id = "78771016858683590931968399206043033368231163700315025308842883779104149970413"

ws = create_connection("wss://ws-subscriptions-clob.polymarket.com/ws/market")

ws.send(json.dumps({
    "type": "market",
    "assets_ids": [token_id]
}))

while True:
    message = ws.recv()
    data = json.loads(message)
    timestamp = datetime.now().isoformat()
    log_entry = f"{timestamp}: {json.dumps(data)}\n"
    
    print(data)
    with open("logs/log_orderbook_python.log", "a") as f:
        f.write(log_entry)
